src/main/java/com/legendary/coffeeShop/service/ComponentService.java

```diff
@@ -0,0 +1,85 @@
+package com.legendary.coffeeShop.service;
+
+import com.legendary.coffeeShop.controller.form.ComponentForm;
+import com.legendary.coffeeShop.dao.entities.Product;
+import com.legendary.coffeeShop.dao.repositories.ComponentRepository;
+import com.legendary.coffeeShop.utils.CommonConstants;
+import com.legendary.coffeeShop.utils.Status;
+import com.legendary.coffeeShop.dao.entities.Component;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class ComponentService {
+
+
+    @Autowired
+    private ComponentRepository componentRepository;
+
+    @Autowired
+    private CommonConstants commonConstants;
+    @Autowired
+    private ProductService productService;
+
+    /*********************************
+     * Public Functions
+     *********************************/
+
+    public Set<Component> getComponentByType(String prodType) {
+        return new HashSet<>(componentRepository.findByProductTypesProductTypeContains(prodType));
+    }
+
+    public Status createComponent(ComponentForm componentForm) {
+        Component component = prepareComponent(new Component(), componentForm);
```

Comment:
Please check if the component exists before you prepared it, also use "getComponent" (you can see an example at productService)